shard-operator-parent/shard-operator-core/src/main/java/com/redhat/service/smartevents/shard/operator/core/providers/IstioGatewayProviderImpl.java

```diff
@@ -17,11 +17,9 @@
 import io.fabric8.kubernetes.api.model.ServicePort;
 import io.fabric8.kubernetes.client.KubernetesClient;
 import io.fabric8.openshift.client.OpenShiftClient;
-import io.quarkus.arc.profile.UnlessBuildProfile;
 import io.quarkus.runtime.StartupEvent;
 
 @ApplicationScoped
-@UnlessBuildProfile(value = "test") // For tests we provide a mocked bean and this should not start at all.
```

Comment:
Is it related with the other change in the properties? I don't understand the need for this change

Reply:
`IstioGatewayProviderImpl.java` class call `setup(@Observes StartupEvent event)` on startup but in testcases we don't want this method to run. We usally mock its behaviour in test. There was no easy way to mock StartupEvent earlier due to which `@UnlessBuildProfile(value = "test")` annotation was added so that bean for this class would not created in `test` profile and MockedBean would be injected.

As I did some research I found that if we add `quarkus.arc.test.disable-application-lifecycle-observers=true` property then testcases would not register for StartupEvent/ShutdownEvent which provide more clearner way because we don't need to add any special handling in code to run the testcase.

If I don't use this new property new I need to do similiar handling in `IstioStartupService` because it also contain `setup(@Observes StartupEvent event)`.

Reply:
Ok but that property, as far as I understand, is disabling similar events for all the classes and not only for `IstioGatewayProviderImpl`. I would prefer to stay with the existing impl so that we are sure to impact only that specific class and moreover we avoid to have to remember to add that property ~everywhere (in this PR you do the change in at least three places so it is a bit more error prone)

Reply:
I have no strong opinion; so happy to follow @danielezonca 

Be aware that use of `quarkus.arc.test.disable-application-lifecycle-observers=true` may not work as expected if we/anyone ever writes tests that use `shard-operator-core` as a dependency vs part of a multi-module project. See https://github.com/quarkusio/quarkus/issues/21487.

Reply:
done.. I had reverted my changes regarding this property and switched back to our old approach.